### PR TITLE
Tab titles now show cwd (#2907)

### LINKED ISSUE #2907: Revisit the idea of rename tabs
- [x] I am on the [latest](https://github.com/zeit/hyper/releases/latest) Hyper.app version
- [x] I have searched the [issues](https://github.com/zeit/hyper/issues) of this repo and believe that this is not a duplicate

- **OS version and name**: OS X El Capitan 10.11.6
- **Hyper.app version**: 2.0.0

## Issue

I know this has been discussed before in #284, but as a SysAdm I frequently have to ssh to other machines and the current approach will fail, because you can no longer rename the tab through bash tricks.

I feel a better way could be the addition of another hyper command(rename-tab) in hyper-keymap and prompt the user in an os-level dialog and ask the new title.

### PR DESCRIPTION
**Pull request by El-Dringo-Brannde**
Closes #2907 #2886 #1188 

This PR is ready to be merged, ran dist build and it works all the same on my Mac. 

**However,** I don't have a PC, or a VM of a PC to test the windows pathing on. Theoretically... It should work as I have kindly borrowed the same code that gets CWD from [hyper-statusline](https://github.com/henrikdahl/hyper-statusline/blob/master/index.js)

### CODE REVIEW

File: lib/utils/fs.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+export default function getCWD(pid, action) {
+  let cwd;
+  if (process.platform == 'win32') {
+    let directoryRegex = /([a-zA-Z]+)/im;
+    if (action && action.data) {
+      let path = directoryRegex.exec(action.data);
+      if (path) cwd = path[0];
+    }
+  } else {
+    cwd = execSync(`lsof -p ${pid} | awk '$4=="cwd"' | tr -s ' ' | cut -d ' ' -f9-`, {encoding: 'utf8'});
```
Comment (albinekb) [inline lib/utils/fs.js]: Is this the best way to get the current CWD? 🤔 I wonder if there is an async way of doing it so we don't lock up the app...

Comment (El-Dringo-Brannde) [reply lib/utils/fs.js]: @albinekb There are async methods of running child_processes, but I thought in Redux making stuff async is frowned upon (still learning, I've heard of thunks being used for async actions). That's why I only have this function run when the user hits the enter key.

Comment (chabou) [reply lib/utils/fs.js]: It will be executed each time that '\n' is output in term. Do not execute `yes` in your fork 😁

**Comment by chabou** [comment]
This is not the right approach.
Performances will be heavily impacted.

BTW I'm using `oh-my-zsh` and my tab title shows my cwd.
But by default, I would have preferred to have current running command. I think that it was the case in hyper 1.X

But feel free to make a plugin with your work 🙏

**Comment by El-Dringo-Brannde** [comment]
If I had it read the `enter` key press to display the current running command would the PR have a chance of being accepted?

**Comment by El-Dringo-Brannde** [comment]
@chabou @albinekb Can I get a re-review, I had it watch on enter key presses only.  Or at the very least a hint on what would be acceptable features for a PR? Trying to get something merged for my final project assignment in my OSS class.

**Comment by Stanzilla** [comment]
@El-Dringo-Brannde willing to pick this up again?

**Pull request closed without merging**